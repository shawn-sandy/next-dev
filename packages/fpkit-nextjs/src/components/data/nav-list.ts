import { MdxFile } from "nextra"
export const navListItems: MdxFile[] = [
    {
        "kind": "MdxPage",
        "name": "articles",
        "route": "/articles",
        "frontMatter": {
            "type": "page",
            "title": "Articles",
            "date": "2020-01-01T00:00:00.000Z",
            "navKey": 2
        }
    },
    {
        "kind": "MdxPage",
        "name": "index",
        "route": "/",
        "frontMatter": {
            "type": "page",
            "title": "About",
            "date": "2020-01-01T00:00:00.000Z",
            "navKey": 3
        }
    },
    {
        "kind": "MdxPage",
        "name": "post",
        "route": "/post",
        "frontMatter": {
            "type": "page",
            "title": "Posts",
            "date": "2020-01-01T00:00:00.000Z",
            "navKey": 1
        }
    }
]