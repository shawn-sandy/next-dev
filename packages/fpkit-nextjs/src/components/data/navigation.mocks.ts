
export const dataSet = [
    {
        kind: "Meta",
        data: {
            index: "Introduction",
            another: "Another Page",
            advanced: "Advanced (A Folder)",
            about: {
                title: "About",
                type: "page",
            },
            contact: {
                title: "Contact â†—",
                type: "page",
                href: "https://twitter.com/shuding_",
                newWindow: true,
            },
        },
    },
    {
        kind: "MdxPage",
        name: "about",
        route: "/about",
    },
    {
        kind: "Folder",
        name: "advanced",
        route: "/advanced",
        children: [
            {
                kind: "MdxPage",
                name: "satori",
                route: "/advanced/satori",
            },
            {
                kind: "Meta",
                data: {
                    satori: "Satori",
                },
            },
        ],
    },
    {
        kind: "MdxPage",
        name: "advanced",
        route: "/advanced",
        frontMatter: {
            title: "Advanced title",
            description:
                "The city bustled with people rushing to work, the sound of honking cars and chatter filling the air. The tall skyscrapers towered over the streets, their windows glinting in the sunlight. A sense of hustle and bustle prevailed in this concrete jungle.",
            date: "2023-01-22T23:51:45.574Z",
        },
    },
    {
        kind: "MdxPage",
        name: "another",
        route: "/another",
        frontMatter: {
            title: "Another page",
            description:
                "Web developers work tirelessly to create seamless digital experiences. They code and debug, troubleshoot and optimize, all to deliver optimal performance and user satisfaction.",
        },
    },
    {
        kind: "MdxPage",
        name: "index",
        route: "/",
        frontMatter: {
            title: "Introduction",
            description:
                "A simple, powerful and flexible site generation framework with everything you love from Next.js.",
        },
    },
];

export const branding = {
    url: "/",
    logo: "Branding Logo",
};
