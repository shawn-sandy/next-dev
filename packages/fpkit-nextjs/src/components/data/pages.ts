

import { Page } from "nextra"

export const pages: Page[] = [
    {
        "kind": "MdxPage",
        "name": "table",
        "route": "/posts/table",
        "frontMatter": {
            "title": "Table",
            "date": "2022/8/28",
            "description": "En example of using table.",
            "tag": "web development",
            "author": "Dimitri POSTOLOV"
        },
        "meta": {
            "title": "Table"
        }
    },
    {
        "kind": "MdxPage",
        "name": "code-blocks",
        "route": "/posts/code-blocks",
        "frontMatter": {
            "title": "Code blocks",
            "date": "2022/7/29",
            "description": "En example of using code blocks in your blog.",
            "tag": "web development,JavaScript,GraphQL,C++,Java,React,Next.js,The Guild,MacBook Pro",
            "author": "Dimitri POSTOLOV"
        },
        "meta": {
            "title": "Code blocks"
        }
    },
    {
        "kind": "MdxPage",
        "name": "index",
        "route": "/posts",
        "frontMatter": {
            "type": "posts",
            "title": "Random Thoughts 1",
            "date": "2020-01-03T00:00:00.000Z"
        },
        "meta": {
            "title": "Random Thoughts 1"
        }
    },
    {
        "kind": "MdxPage",
        "name": "random",
        "route": "/posts/random",
        "frontMatter": {
            "type": "posts",
            "title": "Some Random Thoughts",
            "date": "2020-01-03T00:00:00.000Z"
        },
        "meta": {
            "title": "Some Random Thoughts"
        }
    },
    {
        "kind": "MdxPage",
        "name": "aaron-swartz-a-programmable-web",
        "route": "/posts/aaron-swartz-a-programmable-web",
        "frontMatter": {
            "title": "Notes on A Programmable Web by Aaron Swartz",
            "date": "2016/5/21",
            "description": "At the time when I was getting into web development, I had the chance to read one of the most inspiring book about the web, Aaron Swartz's A Programmable Web. And it completely changed my mind.",
            "tag": "web development",
            "author": "Shu"
        },
        "meta": {
            "title": "Notes on A Programmable Web by Aaron Swartz"
        }
    }
]