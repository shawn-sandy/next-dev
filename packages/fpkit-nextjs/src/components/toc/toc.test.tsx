import React from "react";

import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";

import { headings } from "./toc.test.data";

// ðŸ‘‰ import ComponentName from "./component-name";
import Toc from "./toc";
describe("App renders correctly", () => {
  it("renders without crashing", async () => {
    render(<Toc items={headings} />);
    const toc = screen.getByRole("list");
    expect(toc).toBeDefined();
    // @ts-ignore
    expect(await axe(toc)).toHaveNoViolations();
    expect(toc).toMatchSnapshot();
  });

  it("renders the correct number of items", () => {
    render(<Toc items={headings} />);
    const toc = screen.getByRole("list");
    const links = screen.getAllByRole("link");

    expect(toc.children.length).toBe(headings.length);
    expect(links.length).toBe(headings.length);
  });
});
