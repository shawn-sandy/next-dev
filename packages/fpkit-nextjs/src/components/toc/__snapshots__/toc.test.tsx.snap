// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App renders correctly > renders without crashing 1`] = `
<ul
  data-fp-list="unstyled undefined"
>
  <li>
    <a
      href="#test-filename-line-highlighting-and-empty-lines"
    >
      Test filename, line highlighting and empty lines
    </a>
  </li>
  <li>
    <a
      href="#test-showlinenumbers-and-word-highlighting"
    >
      Test showLineNumbers and word highlighting
    </a>
  </li>
  <li>
    <a
      href="#test-highlighting-inline-code"
    >
      Test highlighting inline code
    </a>
  </li>
  <li>
    <a
      href="#test-without-specified-language"
    >
      Test without specified language
    </a>
  </li>
  <li>
    <a
      href="#test-link-in-code"
    >
      Test link in code
    </a>
  </li>
</ul>
`;
